databases/psql_replication.rst

```diff
@@ -0,0 +1,132 @@
+Postgres Replication
+====================
+
+What to do if Replication Streaming on Postgres fails
+-----------------------------------------------------
+
+Important! Read this section first before doing anything!
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this section, very powerful commands are described. So before execute anything, be 100% sure that you are on the right server!
+Else you could cause a lot of damage. 
+
+To check if you are on the right server use the following query.
+On master the result is empty. On slave it is not (see example). 
+
+#. Check if your on slave
+
+   .. code:: 
+
+       SELECT pg_last_xlog_receive_location();
+       
+         pg_last_xlog_receive_location 
+        -------------------------------
+                2493/54AAF118
+
+Check if Replication is working
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+First of all you should check the delay. If the delay is critical nagios or Backoffice will inform you about it.
+This will look like this:
+
+#. Check the delay
+
+   .. code::
+
+      POSTGRES_HOT_STANDBY_DELAY CRITICAL: DB postgres (host:db01slave) 147866242312 seconds
+
+   If the number (delay in seconds) is extremely high then you know something must be wrong with the replication.
+
+#. Another way to check the delay directly is 
+
+   .. code::
+
+      $ ssh db01a  
+   
+      $ psql -U postgres -h db01slave 
+
+      SELECT now() - pg_last_xact_replay_timestamp() AS delay;
+       
+               delay 
+           --------------
+           04:15:55.03134      
+
+   If the number is very small the slave is probably reading from master in that moment. But always have an look on the issue.
+   If the Number is extremely high then you should do following query:
+
+#. Check the last replay
+
+   .. code::
+
+      SELECT pg_last_xact_replay_timestamp();
+
+        pg_last_xact_replay_timestamp 
+       -------------------------------
+        2017-10-10 11:45:00.945915+02
+
+   It prints the timestamp for the last replication. It is more comprehensive than just a number. 
+   If the delay is more than 3 hours, you should have a look at the postgres log file: /var/log/postgres/postgresql-9.5-main.log.
+   If you find messages like "0000000200002440000000A8: file already removed" then you have to do the basebackup. 
+   Otherwise you could have a look at the postgres configurations. If you cant see anything odd, a restart of postgres could solve the problem.
```

Comment:
You should probably mention that Postgres doesn't replay any changes during dumps (e.g. when creating backups).